src/librustc_codegen_llvm/asm.rs

```diff
@@ -410,10 +413,80 @@ fn inline_asm_call(
     }
 }
 
+/// If the register is an xmm/ymm/zmm register then return its index.
+fn xmm_reg_index(reg: InlineAsmReg) -> Option<u32> {
+    match reg {
+        InlineAsmReg::X86(reg)
+            if reg as u32 >= X86InlineAsmReg::xmm0 as u32
+                && reg as u32 <= X86InlineAsmReg::xmm15 as u32 =>
```

Comment:
```suggestion
        InlineAsmReg::X86(reg)
            if RangeInclusive::new(X86InlineAsmReg::xmm0, X86InlineAsmReg::xmm15).contains(&reg) =>
```

Reply:
Does `RangeInclusive` work on enums?

Reply:
If it implement PartialOrd and/or PartialEq I think.

Reply:
You can just write `(X86InlineAsmReg::xmm0..=X86InlineAsmReg::xmm15).contains(&reg)`, too.